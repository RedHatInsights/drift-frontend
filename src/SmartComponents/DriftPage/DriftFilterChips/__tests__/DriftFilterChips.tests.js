import React from 'react';
import { shallow, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { Provider } from 'react-redux';
import toJson from 'enzyme-to-json';

import ConnectedDriftFilterChips, { DriftFilterChips } from '../DriftFilterChips';
import stateFilters from '../../../modules/__tests__/state-filter.fixtures';

describe('DriftFilterChips', () => {
    it('should render correctly', () =>{
        const props = {
            stateFilters: [],
            factFilter: '' };
        const wrapper = shallow(
            <DriftFilterChips { ...props }/>
        );
        expect(toJson(wrapper)).toMatchSnapshot();
    });
});

describe('ConnectedDriftFilterChips', () => {
    let initialState;
    let mockStore;

    beforeEach(() => {
        mockStore = configureStore();
        initialState = {
            compareState: {
                stateFilters: [],
                factFilter: ''
            }
        };
    });

    it('should render correctly', () => {
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={ store }>
                <ConnectedDriftFilterChips />
            </Provider>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should render all states true', () => {
        initialState.compareState.factFilter = 'something';
        initialState.compareState.stateFilters = stateFilters.allStatesTrue;
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={ store }>
                <ConnectedDriftFilterChips />
            </Provider>
        );

        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('should update stateFilter', () => {
        initialState.compareState.stateFilters = stateFilters.allStatesTrue;
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={ store }>
                <ConnectedDriftFilterChips />
            </Provider>
        );
        wrapper.find('.pf-c-button').first().simulate('click');

        expect(store.getState().compareState.stateFilters).toEqual(stateFilters.sameStateFalse);
    });
});
