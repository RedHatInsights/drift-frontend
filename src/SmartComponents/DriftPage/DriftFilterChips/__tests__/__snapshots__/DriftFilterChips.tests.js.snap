// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConnectedDriftFilterChips should render all states true 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(DriftFilterChips)>
    <DriftFilterChips
      addStateFilter={[Function]}
      factFilter="something"
      filterByFact={[Function]}
      stateFilters={
        Array [
          Object {
            "display": "Same",
            "filter": "SAME",
            "selected": true,
          },
          Object {
            "display": "Different",
            "filter": "DIFFERENT",
            "selected": true,
          },
          Object {
            "display": "Incomplete data",
            "filter": "INCOMPLETE_DATA",
            "selected": true,
          },
        ]
      }
    >
      <ChipGroup
        className=""
        collapsedText="\${remaining} more"
        defaultIsOpen={false}
        expandedText="Show Less"
        numChips={3}
        withToolbar={true}
      >
        <InnerChipGroup
          className=""
          collapsedText="\${remaining} more"
          defaultIsOpen={false}
          expandedText="Show Less"
          isOpen={false}
          numChips={3}
          onToggleCollapse={[Function]}
          withToolbar={true}
        >
          <ChipGroupToolbarItem
            categoryName="State"
            key=".$State"
          >
            <GenerateId
              prefix="pf-random-id-"
            >
              <ul
                className="pf-c-chip-group pf-m-toolbar"
              >
                <li>
                  <h4
                    className="pf-c-chip-group__label"
                    id="pf-random-id-0"
                  >
                    State
                  </h4>
                  <ul
                    className="pf-c-chip-group"
                  >
                    <Component
                      component="li"
                      key=".$State0"
                      onClick={[Function]}
                    >
                      <ComponentWithOuia
                        component={[Function]}
                        componentProps={
                          Object {
                            "children": "Same",
                            "component": "li",
                            "onClick": [Function],
                          }
                        }
                        consumerContext={null}
                      >
                        <Chip
                          className=""
                          closeBtnAriaLabel="close"
                          component="li"
                          isOverflowChip={false}
                          isReadOnly={false}
                          onClick={[Function]}
                          ouiaContext={
                            Object {
                              "isOuia": false,
                              "ouiaId": null,
                            }
                          }
                          tooltipPosition="top"
                        >
                          <GenerateId
                            prefix="pf-random-id-"
                          >
                            <li
                              className="pf-c-chip"
                            >
                              <span
                                className="pf-c-chip__text"
                                id="pf-random-id-1"
                              >
                                Same
                              </span>
                              <ChipButton
                                aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                                ariaLabel="close"
                                id="remove_pf-random-id-1"
                                onClick={[Function]}
                              >
                                <Component
                                  aria-label="close"
                                  aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                                  className=""
                                  id="remove_pf-random-id-1"
                                  onClick={[Function]}
                                  variant="plain"
                                >
                                  <ComponentWithOuia
                                    component={[Function]}
                                    componentProps={
                                      Object {
                                        "aria-label": "close",
                                        "aria-labelledby": "remove_pf-random-id-1 pf-random-id-1",
                                        "children": <TimesCircleIcon
                                          aria-hidden="true"
                                          color="currentColor"
                                          noVerticalAlign={false}
                                          size="sm"
                                          title={null}
                                        />,
                                        "className": "",
                                        "id": "remove_pf-random-id-1",
                                        "onClick": [Function],
                                        "variant": "plain",
                                      }
                                    }
                                    consumerContext={
                                      Object {
                                        "isOuia": false,
                                        "ouiaId": null,
                                      }
                                    }
                                  >
                                    <Button
                                      aria-label="close"
                                      aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                                      className=""
                                      id="remove_pf-random-id-1"
                                      onClick={[Function]}
                                      ouiaContext={
                                        Object {
                                          "isOuia": false,
                                          "ouiaId": null,
                                        }
                                      }
                                      variant="plain"
                                    >
                                      <button
                                        aria-disabled={null}
                                        aria-label="close"
                                        aria-labelledby="remove_pf-random-id-1 pf-random-id-1"
                                        className="pf-c-button pf-m-plain"
                                        disabled={false}
                                        id="remove_pf-random-id-1"
                                        onClick={[Function]}
                                        tabIndex={null}
                                        type="button"
                                      >
                                        <TimesCircleIcon
                                          aria-hidden="true"
                                          color="currentColor"
                                          noVerticalAlign={false}
                                          size="sm"
                                          title={null}
                                        >
                                          <svg
                                            aria-hidden="true"
                                            aria-labelledby={null}
                                            fill="currentColor"
                                            height="1em"
                                            role="img"
                                            style={
                                              Object {
                                                "verticalAlign": "-0.125em",
                                              }
                                            }
                                            viewBox="0 0 512 512"
                                            width="1em"
                                          >
                                            <path
                                              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                                              transform=""
                                            />
                                          </svg>
                                        </TimesCircleIcon>
                                      </button>
                                    </Button>
                                  </ComponentWithOuia>
                                </Component>
                              </ChipButton>
                            </li>
                          </GenerateId>
                        </Chip>
                      </ComponentWithOuia>
                    </Component>
                    <Component
                      component="li"
                      key=".$State1"
                      onClick={[Function]}
                    >
                      <ComponentWithOuia
                        component={[Function]}
                        componentProps={
                          Object {
                            "children": "Different",
                            "component": "li",
                            "onClick": [Function],
                          }
                        }
                        consumerContext={null}
                      >
                        <Chip
                          className=""
                          closeBtnAriaLabel="close"
                          component="li"
                          isOverflowChip={false}
                          isReadOnly={false}
                          onClick={[Function]}
                          ouiaContext={
                            Object {
                              "isOuia": false,
                              "ouiaId": null,
                            }
                          }
                          tooltipPosition="top"
                        >
                          <GenerateId
                            prefix="pf-random-id-"
                          >
                            <li
                              className="pf-c-chip"
                            >
                              <span
                                className="pf-c-chip__text"
                                id="pf-random-id-2"
                              >
                                Different
                              </span>
                              <ChipButton
                                aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                                ariaLabel="close"
                                id="remove_pf-random-id-2"
                                onClick={[Function]}
                              >
                                <Component
                                  aria-label="close"
                                  aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                                  className=""
                                  id="remove_pf-random-id-2"
                                  onClick={[Function]}
                                  variant="plain"
                                >
                                  <ComponentWithOuia
                                    component={[Function]}
                                    componentProps={
                                      Object {
                                        "aria-label": "close",
                                        "aria-labelledby": "remove_pf-random-id-2 pf-random-id-2",
                                        "children": <TimesCircleIcon
                                          aria-hidden="true"
                                          color="currentColor"
                                          noVerticalAlign={false}
                                          size="sm"
                                          title={null}
                                        />,
                                        "className": "",
                                        "id": "remove_pf-random-id-2",
                                        "onClick": [Function],
                                        "variant": "plain",
                                      }
                                    }
                                    consumerContext={
                                      Object {
                                        "isOuia": false,
                                        "ouiaId": null,
                                      }
                                    }
                                  >
                                    <Button
                                      aria-label="close"
                                      aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                                      className=""
                                      id="remove_pf-random-id-2"
                                      onClick={[Function]}
                                      ouiaContext={
                                        Object {
                                          "isOuia": false,
                                          "ouiaId": null,
                                        }
                                      }
                                      variant="plain"
                                    >
                                      <button
                                        aria-disabled={null}
                                        aria-label="close"
                                        aria-labelledby="remove_pf-random-id-2 pf-random-id-2"
                                        className="pf-c-button pf-m-plain"
                                        disabled={false}
                                        id="remove_pf-random-id-2"
                                        onClick={[Function]}
                                        tabIndex={null}
                                        type="button"
                                      >
                                        <TimesCircleIcon
                                          aria-hidden="true"
                                          color="currentColor"
                                          noVerticalAlign={false}
                                          size="sm"
                                          title={null}
                                        >
                                          <svg
                                            aria-hidden="true"
                                            aria-labelledby={null}
                                            fill="currentColor"
                                            height="1em"
                                            role="img"
                                            style={
                                              Object {
                                                "verticalAlign": "-0.125em",
                                              }
                                            }
                                            viewBox="0 0 512 512"
                                            width="1em"
                                          >
                                            <path
                                              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                                              transform=""
                                            />
                                          </svg>
                                        </TimesCircleIcon>
                                      </button>
                                    </Button>
                                  </ComponentWithOuia>
                                </Component>
                              </ChipButton>
                            </li>
                          </GenerateId>
                        </Chip>
                      </ComponentWithOuia>
                    </Component>
                    <Component
                      component="li"
                      key=".$State2"
                      onClick={[Function]}
                    >
                      <ComponentWithOuia
                        component={[Function]}
                        componentProps={
                          Object {
                            "children": "Incomplete data",
                            "component": "li",
                            "onClick": [Function],
                          }
                        }
                        consumerContext={null}
                      >
                        <Chip
                          className=""
                          closeBtnAriaLabel="close"
                          component="li"
                          isOverflowChip={false}
                          isReadOnly={false}
                          onClick={[Function]}
                          ouiaContext={
                            Object {
                              "isOuia": false,
                              "ouiaId": null,
                            }
                          }
                          tooltipPosition="top"
                        >
                          <GenerateId
                            prefix="pf-random-id-"
                          >
                            <li
                              className="pf-c-chip"
                            >
                              <span
                                className="pf-c-chip__text"
                                id="pf-random-id-3"
                              >
                                Incomplete data
                              </span>
                              <ChipButton
                                aria-labelledby="remove_pf-random-id-3 pf-random-id-3"
                                ariaLabel="close"
                                id="remove_pf-random-id-3"
                                onClick={[Function]}
                              >
                                <Component
                                  aria-label="close"
                                  aria-labelledby="remove_pf-random-id-3 pf-random-id-3"
                                  className=""
                                  id="remove_pf-random-id-3"
                                  onClick={[Function]}
                                  variant="plain"
                                >
                                  <ComponentWithOuia
                                    component={[Function]}
                                    componentProps={
                                      Object {
                                        "aria-label": "close",
                                        "aria-labelledby": "remove_pf-random-id-3 pf-random-id-3",
                                        "children": <TimesCircleIcon
                                          aria-hidden="true"
                                          color="currentColor"
                                          noVerticalAlign={false}
                                          size="sm"
                                          title={null}
                                        />,
                                        "className": "",
                                        "id": "remove_pf-random-id-3",
                                        "onClick": [Function],
                                        "variant": "plain",
                                      }
                                    }
                                    consumerContext={
                                      Object {
                                        "isOuia": false,
                                        "ouiaId": null,
                                      }
                                    }
                                  >
                                    <Button
                                      aria-label="close"
                                      aria-labelledby="remove_pf-random-id-3 pf-random-id-3"
                                      className=""
                                      id="remove_pf-random-id-3"
                                      onClick={[Function]}
                                      ouiaContext={
                                        Object {
                                          "isOuia": false,
                                          "ouiaId": null,
                                        }
                                      }
                                      variant="plain"
                                    >
                                      <button
                                        aria-disabled={null}
                                        aria-label="close"
                                        aria-labelledby="remove_pf-random-id-3 pf-random-id-3"
                                        className="pf-c-button pf-m-plain"
                                        disabled={false}
                                        id="remove_pf-random-id-3"
                                        onClick={[Function]}
                                        tabIndex={null}
                                        type="button"
                                      >
                                        <TimesCircleIcon
                                          aria-hidden="true"
                                          color="currentColor"
                                          noVerticalAlign={false}
                                          size="sm"
                                          title={null}
                                        >
                                          <svg
                                            aria-hidden="true"
                                            aria-labelledby={null}
                                            fill="currentColor"
                                            height="1em"
                                            role="img"
                                            style={
                                              Object {
                                                "verticalAlign": "-0.125em",
                                              }
                                            }
                                            viewBox="0 0 512 512"
                                            width="1em"
                                          >
                                            <path
                                              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                                              transform=""
                                            />
                                          </svg>
                                        </TimesCircleIcon>
                                      </button>
                                    </Button>
                                  </ComponentWithOuia>
                                </Component>
                              </ChipButton>
                            </li>
                          </GenerateId>
                        </Chip>
                      </ComponentWithOuia>
                    </Component>
                  </ul>
                </li>
              </ul>
            </GenerateId>
          </ChipGroupToolbarItem>
          <ChipGroupToolbarItem
            categoryName="Fact name"
            key=".$Fact name"
          >
            <GenerateId
              prefix="pf-random-id-"
            >
              <ul
                className="pf-c-chip-group pf-m-toolbar"
              >
                <li>
                  <h4
                    className="pf-c-chip-group__label"
                    id="pf-random-id-4"
                  >
                    Fact name
                  </h4>
                  <ul
                    className="pf-c-chip-group"
                  >
                    <Component
                      component="li"
                      key=".$Fact name3"
                      onClick={[Function]}
                    >
                      <ComponentWithOuia
                        component={[Function]}
                        componentProps={
                          Object {
                            "children": "something",
                            "component": "li",
                            "onClick": [Function],
                          }
                        }
                        consumerContext={null}
                      >
                        <Chip
                          className=""
                          closeBtnAriaLabel="close"
                          component="li"
                          isOverflowChip={false}
                          isReadOnly={false}
                          onClick={[Function]}
                          ouiaContext={
                            Object {
                              "isOuia": false,
                              "ouiaId": null,
                            }
                          }
                          tooltipPosition="top"
                        >
                          <GenerateId
                            prefix="pf-random-id-"
                          >
                            <li
                              className="pf-c-chip"
                            >
                              <span
                                className="pf-c-chip__text"
                                id="pf-random-id-5"
                              >
                                something
                              </span>
                              <ChipButton
                                aria-labelledby="remove_pf-random-id-5 pf-random-id-5"
                                ariaLabel="close"
                                id="remove_pf-random-id-5"
                                onClick={[Function]}
                              >
                                <Component
                                  aria-label="close"
                                  aria-labelledby="remove_pf-random-id-5 pf-random-id-5"
                                  className=""
                                  id="remove_pf-random-id-5"
                                  onClick={[Function]}
                                  variant="plain"
                                >
                                  <ComponentWithOuia
                                    component={[Function]}
                                    componentProps={
                                      Object {
                                        "aria-label": "close",
                                        "aria-labelledby": "remove_pf-random-id-5 pf-random-id-5",
                                        "children": <TimesCircleIcon
                                          aria-hidden="true"
                                          color="currentColor"
                                          noVerticalAlign={false}
                                          size="sm"
                                          title={null}
                                        />,
                                        "className": "",
                                        "id": "remove_pf-random-id-5",
                                        "onClick": [Function],
                                        "variant": "plain",
                                      }
                                    }
                                    consumerContext={
                                      Object {
                                        "isOuia": false,
                                        "ouiaId": null,
                                      }
                                    }
                                  >
                                    <Button
                                      aria-label="close"
                                      aria-labelledby="remove_pf-random-id-5 pf-random-id-5"
                                      className=""
                                      id="remove_pf-random-id-5"
                                      onClick={[Function]}
                                      ouiaContext={
                                        Object {
                                          "isOuia": false,
                                          "ouiaId": null,
                                        }
                                      }
                                      variant="plain"
                                    >
                                      <button
                                        aria-disabled={null}
                                        aria-label="close"
                                        aria-labelledby="remove_pf-random-id-5 pf-random-id-5"
                                        className="pf-c-button pf-m-plain"
                                        disabled={false}
                                        id="remove_pf-random-id-5"
                                        onClick={[Function]}
                                        tabIndex={null}
                                        type="button"
                                      >
                                        <TimesCircleIcon
                                          aria-hidden="true"
                                          color="currentColor"
                                          noVerticalAlign={false}
                                          size="sm"
                                          title={null}
                                        >
                                          <svg
                                            aria-hidden="true"
                                            aria-labelledby={null}
                                            fill="currentColor"
                                            height="1em"
                                            role="img"
                                            style={
                                              Object {
                                                "verticalAlign": "-0.125em",
                                              }
                                            }
                                            viewBox="0 0 512 512"
                                            width="1em"
                                          >
                                            <path
                                              d="M256 8C119 8 8 119 8 256s111 248 248 248 248-111 248-248S393 8 256 8zm121.6 313.1c4.7 4.7 4.7 12.3 0 17L338 377.6c-4.7 4.7-12.3 4.7-17 0L256 312l-65.1 65.6c-4.7 4.7-12.3 4.7-17 0L134.4 338c-4.7-4.7-4.7-12.3 0-17l65.6-65-65.6-65.1c-4.7-4.7-4.7-12.3 0-17l39.6-39.6c4.7-4.7 12.3-4.7 17 0l65 65.7 65.1-65.6c4.7-4.7 12.3-4.7 17 0l39.6 39.6c4.7 4.7 4.7 12.3 0 17L312 256l65.6 65.1z"
                                              transform=""
                                            />
                                          </svg>
                                        </TimesCircleIcon>
                                      </button>
                                    </Button>
                                  </ComponentWithOuia>
                                </Component>
                              </ChipButton>
                            </li>
                          </GenerateId>
                        </Chip>
                      </ComponentWithOuia>
                    </Component>
                  </ul>
                </li>
              </ul>
            </GenerateId>
          </ChipGroupToolbarItem>
        </InnerChipGroup>
      </ChipGroup>
    </DriftFilterChips>
  </Connect(DriftFilterChips)>
</Provider>
`;

exports[`ConnectedDriftFilterChips should render correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(DriftFilterChips)>
    <DriftFilterChips
      addStateFilter={[Function]}
      factFilter=""
      filterByFact={[Function]}
      stateFilters={Array []}
    >
      <ChipGroup
        className=""
        collapsedText="\${remaining} more"
        defaultIsOpen={false}
        expandedText="Show Less"
        numChips={3}
        withToolbar={true}
      >
        <InnerChipGroup
          className=""
          collapsedText="\${remaining} more"
          defaultIsOpen={false}
          expandedText="Show Less"
          isOpen={false}
          numChips={3}
          onToggleCollapse={[Function]}
          withToolbar={true}
        >
          <ChipGroupToolbarItem
            categoryName="State"
            key=".$State"
          />
          <ChipGroupToolbarItem
            categoryName="Fact name"
            key=".$Fact name"
          />
        </InnerChipGroup>
      </ChipGroup>
    </DriftFilterChips>
  </Connect(DriftFilterChips)>
</Provider>
`;

exports[`DriftFilterChips should render correctly 1`] = `
<ChipGroup
  className=""
  collapsedText="\${remaining} more"
  defaultIsOpen={false}
  expandedText="Show Less"
  numChips={3}
  withToolbar={true}
>
  <ChipGroupToolbarItem
    categoryName="State"
    key="State"
  />
  <ChipGroupToolbarItem
    categoryName="Fact name"
    key="Fact name"
  />
</ChipGroup>
`;
